File: frontend/src/types/lopa.ts
export interface DocumentMonthlyBatchCreate {
	doc_type: string;
	reporting_year: number;
	reporting_month: number;
	mode?: "append" | "overwrite";
	items: Record<string, any>[];
}

export interface LopaHistoryResponse {
	upload_batch_id: string;
	reporting_year: number;
	reporting_month: number;
	upload_date: string;
	record_count: number;
}

export interface LopaDataResponse {
	id: string;
	upload_batch_id: string;
	owner_account_id: string | null;
	reporting_year: number;
	reporting_month: number;
	function_no: string | null;
	function_name: string | null;
	function_description: string | null;
	final_element: string | null;
	recommendation: string | null;
	rrf_gap_value: string | null;
	rrf_gap_type: string | null;
	responsibility_pic: string | null;
	target_date: string | null;
	reminder_status: string | null;
	response_progress: string | null;
	status: string | null;
	evidence: string | null;
	completion_date: string | null;
	created_at: string;
	updated_at: string | null;
}

File: backend/src/models/schemas/lopa.py
import typing
import datetime
import pydantic
from src.models.schemas.location import FieldLocationResponse

class DocumentMonthlyBatchCreate(pydantic.BaseModel):
    doc_type: str
    reporting_year: int
    reporting_month: int
    field: str | None = None
    mode: str = "append"
    items: list[dict[str, typing.Any]]

    class Config:
        allow_population_by_field_name = True
        orm_mode = True

class LopaResponse(pydantic.BaseModel):
    id: typing.Any
    upload_batch_id: typing.Any
    owner_account_id: typing.Any | None
    reporting_year: int
    reporting_month: int
    field: str | None
    field_location: FieldLocationResponse | None = None
    function_no: str | None
    function_name: str | None
    function_description: str | None
    final_element: str | None
    recommendation: str | None
    rrf_gap_value: str | None
    rrf_gap_type: str | None
    responsibility_pic: str | None
    target_date: datetime.date | None
    reminder_status: str | None
    response_progress: str | None
    status: str | None
    evidence: str | None
    completion_date: datetime.date | None
    created_at: datetime.datetime
    updated_at: datetime.datetime | None

    class Config:
        orm_mode = True

class LopaHistoryResponse(pydantic.BaseModel):
    upload_batch_id: typing.Any
    reporting_year: int
    reporting_month: int
    field: str | None
    upload_date: datetime.datetime
    record_count: int
    
    class Config:
        orm_mode = True
